owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -640,16 +644,17 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
     }
 
     private fun initBrandableOptionsUI() {
-        if (!contextProvider.getBoolean(R.bool.show_server_url_input)) {
-            binding.hostUrlFrame.isVisible = false
-            binding.centeredRefreshButton.run {
-                isVisible = true
-                setOnClickListener { checkOcServer() }
-            }
+        val preferencesProvider = SharedPreferencesProviderImpl(this)
+        val showInput = if (BuildConfig.FLAVOR == MDM_FLAVOR) preferencesProvider.getBoolean(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY, contextProvider.getBoolean(R.bool.show_server_url_input)) else contextProvider.getBoolean(R.bool.show_server_url_input)
+        binding.hostUrlFrame.isVisible = showInput
+        binding.centeredRefreshButton.isVisible = !showInput
+        if (!showInput) {
+            binding.centeredRefreshButton.setOnClickListener { checkOcServer() }
         }
 
-        if (contextProvider.getString(R.string.server_url).isNotEmpty()) {
-            binding.hostUrlInput.setText(contextProvider.getString(R.string.server_url))
+        val url = if (BuildConfig.FLAVOR == MDM_FLAVOR) preferencesProvider.getString(CONFIGURATION_SERVER_URL, contextProvider.getString(R.string.server_url)) else contextProvider.getString(R.string.server_url)
```

Comment:
This logic shouldn't belong to the activity, but we can keep it at least for the moment till we add more mdm parameters 👍